source/transfer_list.rst

```diff
@@ -146,6 +146,221 @@ The TE header is defined in :numref:`tab_te_header`.
      - The exact size of the data content in bytes, not including inter-TE padding. May be 0.
 
 
+.. _sec_operations:
+
+Standard operations
+-------------------
+
+This section describes the valid operations that can be performed on a TL in
+more detail, in order to clarify how to use the various fields and to serve as a
+guideline for implementation.
+
+Validating a TL header
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. default-role:: code
+
+Inputs:
+
+- `tl_base_addr`: Base address of the existing TL.
+
+#. Compare `tl.signature` (`tl_base_addr + 0x0`) to `0x6e_d0ff`. On a mismatch,
+   abort (this is not a valid TL).
+
+#. Compare `tl.version` (`tl_base_addr + 0x5`) to the expected version
+   (currently 1). On a 0, abort (this is not a valid TL). On any other mismatch
+   (i.e. larger than 1), the TL is valid for reading but must not be modified
+   or relocated.
+
+#. *(optional)* Check that `tl.size` (`tl_base_addr + 0x8`) is smaller or equal
+   to `tl.max_size` (`tl_base_addr + 0xc`), and that `tl.max_size` is smaller or
+   equal to the size of the total area reserved for the TL (if known). If not,
+   abort (TL is corrupted).
+
+#. *(optional)* Check that the sum of `tl.size` bytes starting at `tl_base_addr`
+   minus `tl.checksum` is equal to `tl.checksum`. If not, abort (TL is corrupted).
+
+Reading a TL
+^^^^^^^^^^^^
+
+Inputs:
+
+- `tl_base_addr`: Base address of the existing TL.
+
+#. Calculate `te_base_addr` as `align8(tl_base_addr + tl.hdr_size)`. (Do not
+   hardcode the value for `tl.hdr_size`!)
+
+#. While `te_base_addr - tl_base_addr` is smaller or equal to `tl.size`:
+
+   #. *(optional)* Check that `te_base_addr + te.hdr_size + te.data_size - tl_base_addr`
+      is smaller or equal to `tl.size`, otherwise abort (the TL is corrupted).
+
+   #. If `te.tag_id` (`te_base_addr + 0x0`) is a known tag, interpret the data
+      at `te_base_addr + te.hdr_size` accordingly. (Do not hardcode the value
+      for `te.hdr_size`!)
+
+   #. Add `te.hdr_size + te.data_size` to `te_base_addr`.
```

Comment:
Is this line meant to state how to compute the next TE address?
Should it say something like: "The next TE is at align8(te.hdr_size + te.data_size + te_base_addr)"
?

Reply:
The whole thing is supposed to be a while loop in pseudocode. The indented part is the inner part of the loop. Do you think that's not clear enough (would it help to put a `#. Repeat` at the end or something like that)?

Reply:
I understood it now, we're accumulating in te_base_addr.
it's effectively te_base_addr = te_base_addr + te.hdr_size + te.data_size.

Looks good to me!